src/test/java/com/google/devtools/build/lib/remote/SimpleBlobStoreFactoryTest.java

```diff
@@ -0,0 +1,190 @@
+package com.google.devtools.build.lib.remote;
+
+import com.google.devtools.build.lib.remote.blobstore.CombinedDiskHttpBlobStore;
+import com.google.devtools.build.lib.remote.blobstore.OnDiskBlobStore;
+import com.google.devtools.build.lib.remote.blobstore.SimpleBlobStore;
+import com.google.devtools.build.lib.remote.blobstore.http.HttpBlobStore;
+import com.google.devtools.build.lib.vfs.Path;
+import com.google.devtools.build.lib.vfs.PathFragment;
+import com.google.devtools.common.options.Options;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.IOException;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+@RunWith(JUnit4.class)
+public class SimpleBlobStoreFactoryTest {
+
+  @Rule
+  public ExpectedException expectedEx = ExpectedException.none();
+
+  @Mock
+  private Path workingDirectory;
+
+  @Mock
+  private Path relativeDirectory;
+
+  @Before
+  public final void setUp() {
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @Test
+  public void createCombinedCache() throws IOException {
+    RemoteOptions options = Options.getDefaults(RemoteOptions.class);
+    options.remoteHttpCache = "http://doesnotesist.com";
+    options.diskCache = PathFragment.create("/etc/something/cache/here");
+
+    when(workingDirectory.getRelative(options.diskCache)).thenReturn(relativeDirectory);
```

Comment:
`/etc/something/cache/here` is an absolute path. An interesting case to test would be to test behaviors when `diskCache` is absolute vs relative.

Reply:
agree :) But this would be a test case for a `workingDirectory.getRelative()` method already, which is out of scope for this change (otherwise I will end-up refactoring/writing tests for half of bazel :D )